src/core/server/http/security_response_headers_config.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+const strictTransportSecuritySchema = schema.object({
+  // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+  maxAge: schema.duration({ defaultValue: '1Y' }),
+  includeSubDomains: schema.boolean({ defaultValue: false }),
+  preload: schema.boolean({ defaultValue: false }),
+});
+
+export const securityResponseHeadersSchema = schema.object({
+  strictTransportSecurity: schema.oneOf([strictTransportSecuritySchema, schema.literal(null)], {
+    defaultValue: strictTransportSecuritySchema.validate({}),
+  }),
+  xContentTypeOptions: schema.oneOf([schema.literal('nosniff'), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+    defaultValue: 'nosniff',
+  }),
```

Comment:
By default, `X-Content-Type-Options` is enabled. It only has one valid value. This shouldn't have any negative impact on users.